src/postgres/src/backend/tcop/postgres.c

```diff
@@ -5129,6 +5139,73 @@ PostgresMain(int argc, char *argv[],
 	if (!ignore_till_sync)
 		send_ready_for_query = true;	/* initially, or after error */
 
+	int	num_backends = pgstat_fetch_stat_numbackends();
+	for (int curr_backend = 1; curr_backend <= num_backends; curr_backend++)
+	{
+		PgBackendStatus *beentry = pgstat_fetch_stat_beentry(curr_backend);
+		SockAddr	zero_clientaddr;
+		char		remote_host[NI_MAXHOST];
+		char		remote_port[NI_MAXSERV];
+		remote_host[0] = '\0';
+		remote_port[0] = '\0';
+
+		if (beentry->st_procpid != MyProcPid)
+			continue;
+		
+		/* A zeroed client addr means we don't know */
+		memset(&zero_clientaddr, 0, sizeof(zero_clientaddr));
+		if (memcmp(&(beentry->st_clientaddr), &zero_clientaddr,
+					sizeof(zero_clientaddr)) == 0)
+		{
+			ereport(LOG, (errmsg("PG client ip and port not found")));
+		}
+		else
+		{
+			if (beentry->st_clientaddr.addr.ss_family == AF_INET
+#ifdef HAVE_IPV6
+				|| beentry->st_clientaddr.addr.ss_family == AF_INET6
+#endif
+				)
+			{
+
+				int ret = pg_getnameinfo_all(&beentry->st_clientaddr.addr,
+											beentry->st_clientaddr.salen,
+											remote_host, sizeof(remote_host),
+											remote_port, sizeof(remote_port),
+											NI_NUMERICHOST | NI_NUMERICSERV);
+
+				if (ret == 0)
+				{
+					clean_ipv6_addr(beentry->st_clientaddr.addr.ss_family, remote_host);
+				}
+				else
+				{
+					ereport(LOG, (errmsg("PG client ip and port not found")));
+				}
+			}
+			else if (beentry->st_clientaddr.addr.ss_family == AF_UNIX)
+			{
+					/*
+						* Unix sockets always reports NULL for host and -1 for
+						* port, so it's possible to tell the difference to
+						* connections we have no permissions to view, or with
+						* errors.
+						*/
+					ereport(LOG, (errmsg("PG client ip and port not found")));
+			}
+			else
+			{
+				/* Unknown address type, should never happen */
+				ereport(LOG, (errmsg("PG client ip and port not found")));
+			}
+		}
+
+		if (IsYugaByteEnabled())
+			HandleYBStatus(YBCPgSetAUHMetadata(remote_host, atoi(remote_port)));
+		else
+		 	ereport(LOG, (errmsg("yugabyte not enabled yet")));
```

Comment:
Is this required? I think there are multiple places where the code moves on silently if not run as YB